graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlErrorsHandlerTest.java

```diff
@@ -0,0 +1,112 @@
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorException;
+import graphql.schema.DataFetcher;
+
+public class GraphqlErrorsHandlerTest {
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/test.graphqls").toURI());
+
+            final GraphqlErrorsHandler errorsHandler = new GraphqlErrorsHandler() {
+                @Override
+                public @Nullable HttpResponse handle(ServiceRequestContext ctx, ExecutionInput input,
+                                                     ExecutionResult result, MediaType negotiatedProduceType,
+                                                     @Nullable Throwable cause) {
+                    final List<GraphQLError> errors = result.getErrors();
+                    if (errors.stream().map(GraphQLError::getMessage).anyMatch(m -> m.endsWith("foo"))) {
+                        return HttpResponse.ofJson(HttpStatus.OK);
+                    }
+                    return null;
+                }
+            };
+
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> foo = dataFetcher("foo");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", foo));
+                                      final DataFetcher<String> error = dataFetcher("error");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("error", error));
+                                  })
+                                  .errorsHandler(errorsHandler)
+                                  .build();
+            sb.service("/graphql", service);
+        }
+    };
+
+    private static DataFetcher<String> dataFetcher(String value) {
+        return environment -> {
+            final ServiceRequestContext ctx = GraphqlServiceContexts.get(environment);
+            assertThat(ctx.eventLoop().inEventLoop()).isTrue();
+            // Make sure that a ServiceRequestContext is available
+            assertThat(ServiceRequestContext.current()).isSameAs(ctx);
+            throw GraphqlErrorException.newErrorException().message(value).build();
+        };
+    }
+
+    @Test
+    void shouldHandleFooError() {
```

Comment:
I think foo error doesn't mean anything. 😄
How about renaming this to `handledError` and `unhandledError` for the below method?